Sources/UI/Chat/Chat View Controller/ChatViewController+Composer.swift

```diff
@@ -456,36 +456,42 @@ extension ChatViewController {
             return
         }
         
-        showImagePicker(sourceType: pickerSourceType) { [weak self] pickedImage, status in
+        showImagePicker(sourceType: pickerSourceType) { [weak self] result in
             guard let self = self else {
                 return
             }
             
-            guard status == .authorized else {
-                self.showImagePickerAuthorizationStatusAlert(status)
-                return
-            }
-            
-            if let resources = try? pickedImage?.fileURL?.resourceValues(forKeys: [.fileSizeKey]),
-               let fileSize = resources.fileSize,
-               fileSize >= 20 * 1_048_576 { // 20 MB Upload limit
-                self.show(errorMessage: "File size exceeds limit of 20MB")
-                return
-            }
-            
-            guard let presenter = self.presenter,
-                let pickedImage = pickedImage,
-                (pickedImage.fileURL != nil || pickedImage.image != nil) else {
-                return
+            do {
+                let pickedImage = try result.get()
+                
+                if let resources = try? pickedImage.fileURL?.resourceValues(forKeys: [.fileSizeKey]) {
+                    if let fileSize = resources.fileSize,
+                       fileSize >= 20 * 1_048_576 { // 20 MB Upload limit
+                        self.show(errorMessage: "File size exceeds limit of 20MB")
+                        return
+                    }
+                } else {
+                    ClientLogger.log("📁", "File size limit cannot be determined. "
```

Comment:
@VojtaStavik here

Reply:
Ah, I though, you would replace `try?` on L467 with `try` and use the existing `do { } catch { }` to actually get the error and log it.

Reply:
I think it's good to show the error as well as a reason why the file size wasn't determined.